### ISSUE TITLE
Unable to build unter RHEL 9.4

### ISSUE THREAD

**Issue by tuxmaster5000** [Bioconductor/S4Arrays#6]
The build fails with:
```
* installing *source* package ‘S4Arrays’ ...
** using staged installation
** libs
using C compiler: ‘gcc (GCC) 11.4.1 20231218 (Red Hat 11.4.1-3)’
gcc -I"/usr/include/R" -DNDEBUG  -I'/home/foo/R/x86_64-redhat-linux-gnu-library/4.4/S4Vectors/include' -I/usr/local/include    -fpic  -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m64 -march=x86-64-v2 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection  -c R_init_S4Arrays.c -o R_init_S4Arrays.o
gcc -I"/usr/include/R" -DNDEBUG  -I'/home/foo/R/x86_64-redhat-linux-gnu-library/4.4/S4Vectors/include' -I/usr/local/include    -fpic  -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m64 -march=x86-64-v2 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection  -c S4Vectors_stubs.c -o S4Vectors_stubs.o
gcc -I"/usr/include/R" -DNDEBUG  -I'/home/foo/R/x86_64-redhat-linux-gnu-library/4.4/S4Vectors/include' -I/usr/local/include    -fpic  -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m64 -march=x86-64-v2 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection  -c abind.c -o abind.o
gcc -I"/usr/include/R" -DNDEBUG  -I'/home/foo/R/x86_64-redhat-linux-gnu-library/4.4/S4Vectors/include' -I/usr/local/include    -fpic  -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m64 -march=x86-64-v2 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection  -c array_selection.c -o array_selection.o
array_selection.c: In function ‘C_Lindex2Mindex’:
array_selection.c:352:17: error: format not a string literal and no format arguments [-Werror=format-security]
  352 |                 error(errmsg_buf());
      |                 ^~~~~
array_selection.c: In function ‘C_Mindex2Lindex’:
array_selection.c:416:17: error: format not a string literal and no format arguments [-Werror=format-security]
  416 |                 error(errmsg_buf());
      |                 ^~~~~
cc1: some warnings being treated as errors
make: *** [/usr/lib64/R/etc/Makeconf:195: array_selection.o] Error 1
ERROR: compilation failed for package ‘S4Arrays’
```
Using R 4.4. on RHEL 9.4

**Comment by hpages** [comment]
This was fixed in January in **S4Arrays** 1.3.2 (see commit ed420e93e484e9d096833dd8aa7b0496008a0336) but it looks like somehow you are trying to install an older version. What do you get when you run `BiocManager::valid()` in R?

**S4Arrays** is a Bioconductor package (see https://bioconductor.org/packages/S4Arrays), and like any other Bioconductor package, it should _always_ be installed via `BiocManager::install()`. This is the only way to guarantee that you'll get the version of Bioconductor that corresponds to your version of R.

Assuming that you're using R 4.4, this will get you the latest BioC version (3.19, released a couple of weeks ago) which includes the latest **S4Arrays** (version 1.4.0).

Best,
H.

**Comment by tuxmaster5000** [comment]
Sorry my mistake. Update it to 1.4.0 will fix it. I have uses an old bioconductor, which was for R 4.3 instant of 4.4.
